string path
float64 leaf_size
float64 smooth_factor
---